mesonbuild/backend/xcodebackend.py

```diff
@@ -1219,6 +1219,8 @@ def generate_generator_target_shell_build_phases(self, objects_dict):
                     generator_id += 1
 
     def generate_single_generator_phase(self, tname, t, genlist, generator_id, objects_dict):
+        # TODO: this should be rewritten to use the meson wrapper, like the other generators do
+        # Currently it doesn't handle a host binary that requires an exe wrapper correctly.
```

Comment:
Can we just use get_executable_serialisation here, to lower it to strings? Or maybe just use as_meson_exe_cmdline? It's not entirely clear to me what this code does either, but at some point it definitely seems to be using just a string list of commands to run, which is good I think?

Reply:
Likely we can, but I'm not willing to make any changes that I can't test, and I don't have a mac anymore, and we don't test xcode in CI, so...

Reply:
... fair point, I guess.